src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -879,4 +879,22 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_RestoreFallbackFoldersOverride>$(RestoreFallbackFolders)</_RestoreFallbackFoldersOverride>
     </PropertyGroup>
   </Target>
+
+  <!--
+    ============================================================
+    _CleanCacheFiles
+    Deletes all the *.nuget.cache files in the obj directory
+    ============================================================
+  -->
+  <Target Name="_CleanCacheFiles" 
+          AfterTargets="Clean">
+    <PropertyGroup>
+        <RestoreOutputPath Condition=" '$(RestoreOutputPath)' == '' ">$(BaseIntermediateOutputPath)</RestoreOutputPath>
```

Comment:
maybe use another property instead of RestoreOutputPath in order to minimize unintended consequences of setting it here?